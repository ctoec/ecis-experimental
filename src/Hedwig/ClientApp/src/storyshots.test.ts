import initStoryshots from '@storybook/addon-storyshots';

jest.mock('react-dates', () => ({
	// https://github.com/airbnb/react-dates/issues/528
	DayPickerSingleDateController: ({ ...props }) =>
		`<DayPickerSingleDateController" ${JSON.stringify(props)}/>`,
}));

initStoryshots();

jest.resetAllMocks();
