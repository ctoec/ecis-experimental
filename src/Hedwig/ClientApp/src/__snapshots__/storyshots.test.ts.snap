// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Button Attention needed 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <span
    className="oec-inline-icon oec-inline-icon--attentionNeeded"
  >
    <span
      className="usa-sr-only"
    >
      (
      attention needed
      )
    </span>
  </span>
</div>
`;

exports[`Storyshots Button Base 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <button
    className="usa-button usa-button--base"
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;

exports[`Storyshots Button Default 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <button
    className="usa-button "
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;

exports[`Storyshots Button Disabled 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <button
    className="usa-button "
    disabled={true}
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;

exports[`Storyshots Button Secondary 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <button
    className="usa-button usa-button--secondary"
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;

exports[`Storyshots Header Logged in 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <div
    className=""
  >
    <div
      className="usa-overlay"
      onClick={[Function]}
    />
    <header
      className="usa-header usa-header--extended"
      role="banner"
    >
      <div
        className="usa-navbar"
      >
        <div
          className="usa-logo"
          id="extended-logo"
        >
          <em
            className="usa-logo__text"
          >
            <a
              aria-label="Home"
              href="/"
              onClick={[Function]}
              title="Home"
            >
              Hello world!
            </a>
          </em>
        </div>
        <button
          className="usa-menu-btn"
          onClick={[Function]}
        >
          Menu
        </button>
      </div>
      <nav
        className="usa-nav"
        role="navigation"
      >
        <div
          className="usa-nav__inner"
        >
          <button
            className="usa-nav__close"
            onClick={[Function]}
          >
            <img
              alt="close"
              src="close.svg"
            />
          </button>
          <ul
            className="usa-nav__primary usa-accordion"
          >
            <li
              className="usa-nav__primary-item"
            >
              <a
                className="usa-nav__link usa-current"
                href="/"
                onClick={[Function]}
              >
                <span>
                  Active section
                </span>
              </a>
            </li>
            <li
              className="usa-nav__primary-item"
            >
              <a
                className="usa-nav__link"
                href="/another"
                onClick={[Function]}
              >
                <span>
                  Another section
                </span>
              </a>
            </li>
            <li
              className="usa-nav__primary-item"
            >
              <a
                className="usa-nav__link"
                href="/attention"
                onClick={[Function]}
              >
                <span>
                  Attention needed
                  <span
                    className="attention-needed"
                  >
                     •
                    <span
                      className="usa-sr-only"
                    >
                       (attention needed)
                    </span>
                  </span>
                </span>
              </a>
            </li>
          </ul>
          <div
            className="usa-nav__secondary"
          >
            <ul
              className="usa-nav__secondary-links"
            >
              <li
                className="usa-nav__secondary-item"
              >
                <a
                  href="/secondary"
                  onClick={[Function]}
                >
                  Secondary item
                </a>
              </li>
              <li
                className="usa-nav__secondary-item"
              >
                <a
                  href="/secondary2"
                  onClick={[Function]}
                >
                  Another secondary item
                </a>
              </li>
              <li
                className="usa-nav__secondary-item"
              >
                <a
                  href="/logout"
                  onClick={[Function]}
                >
                  Log out
                </a>
              </li>
            </ul>
            <div
              className="oec-logged-in-user"
            >
              <span>
                Hi, 
                Chris
                .
              </span>
            </div>
          </div>
        </div>
      </nav>
    </header>
  </div>
</div>
`;

exports[`Storyshots Header Logged out 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <div
    className=""
  >
    <div
      className="usa-overlay"
      onClick={[Function]}
    />
    <header
      className="usa-header usa-header--extended"
      role="banner"
    >
      <div
        className="usa-navbar"
      >
        <div
          className="usa-logo"
          id="extended-logo"
        >
          <em
            className="usa-logo__text"
          >
            <a
              aria-label="Home"
              href="/"
              onClick={[Function]}
              title="Home"
            >
              Hello world!
            </a>
          </em>
        </div>
        <button
          className="usa-menu-btn"
          onClick={[Function]}
        >
          Menu
        </button>
      </div>
      <nav
        className="usa-nav"
        role="navigation"
      >
        <div
          className="usa-nav__inner"
        >
          <button
            className="usa-nav__close"
            onClick={[Function]}
          >
            <img
              alt="close"
              src="close.svg"
            />
          </button>
          <ul
            className="usa-nav__primary usa-accordion"
          >
            <li
              className="usa-nav__primary-item"
            >
              <a
                className="usa-nav__link usa-current"
                href="/"
                onClick={[Function]}
              >
                <span>
                  Active section
                </span>
              </a>
            </li>
            <li
              className="usa-nav__primary-item"
            >
              <a
                className="usa-nav__link"
                href="/another"
                onClick={[Function]}
              >
                <span>
                  Another section
                </span>
              </a>
            </li>
            <li
              className="usa-nav__primary-item"
            >
              <a
                className="usa-nav__link"
                href="/attention"
                onClick={[Function]}
              >
                <span>
                  Attention needed
                  <span
                    className="attention-needed"
                  >
                     •
                    <span
                      className="usa-sr-only"
                    >
                       (attention needed)
                    </span>
                  </span>
                </span>
              </a>
            </li>
          </ul>
          <div
            className="usa-nav__secondary"
          >
            <ul
              className="usa-nav__secondary-links"
            >
              <li
                className="usa-nav__secondary-item"
              >
                <a
                  href="/secondary"
                  onClick={[Function]}
                >
                  Secondary item
                </a>
              </li>
              <li
                className="usa-nav__secondary-item"
              >
                <a
                  href="/secondary2"
                  onClick={[Function]}
                >
                  Another secondary item
                </a>
              </li>
            </ul>
            <div
              className="oec-logged-in-user"
            >
              <a
                href="/login"
                onClick={[Function]}
              >
                Log in
              </a>
            </div>
          </div>
        </div>
      </nav>
    </header>
  </div>
</div>
`;

exports[`Storyshots Table Full width 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <table
    className="oec-table oec-table--full-width"
    id="my-table"
  >
    <thead>
      <tr>
        <th
          className="oec-table__column-header oec-sortable"
          scope="col"
        >
          <span
            className="oec-table__column-title"
            onClick={[Function]}
          >
            Name
          </span>
          <div
            className="oec-table__sort-controls"
          >
            <input
              checked={true}
              id="my-table-col0-asc"
              name="my-table-column-0-sort"
              onChange={[Function]}
              type="checkbox"
              value="asc"
            />
            <label
              className="oec-table__sort-controls__asc"
              htmlFor="my-table-col0-asc"
            >
              <span>
                Sort ascending
              </span>
            </label>
            <input
              checked={false}
              id="my-table-col0-desc"
              name="my-table-column-0-sort"
              onChange={[Function]}
              type="checkbox"
              value="desc"
            />
            <label
              className="oec-table__sort-controls__desc"
              htmlFor="my-table-col0-desc"
            >
              <span>
                Sort descending
              </span>
            </label>
          </div>
        </th>
        <th
          className="oec-table__column-header oec-sortable"
          scope="col"
        >
          <span
            className="oec-table__column-title"
            onClick={[Function]}
          >
            Price
          </span>
          <div
            className="oec-table__sort-controls"
          >
            <input
              checked={false}
              id="my-table-col1-asc"
              name="my-table-column-1-sort"
              onChange={[Function]}
              type="checkbox"
              value="asc"
            />
            <label
              className="oec-table__sort-controls__asc"
              htmlFor="my-table-col1-asc"
            >
              <span>
                Sort ascending
              </span>
            </label>
            <input
              checked={false}
              id="my-table-col1-desc"
              name="my-table-column-1-sort"
              onChange={[Function]}
              type="checkbox"
              value="desc"
            />
            <label
              className="oec-table__sort-controls__desc"
              htmlFor="my-table-col1-desc"
            >
              <span>
                Sort descending
              </span>
            </label>
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          Apple
        </th>
        <th
          scope="row"
        >
          $0.75
        </th>
      </tr>
      <tr>
        <th
          scope="row"
        >
          Avocado
        </th>
        <th
          scope="row"
        >
          $1.80
        </th>
      </tr>
      <tr>
        <th
          scope="row"
        >
          Pear
        </th>
        <th
          scope="row"
        >
          $1.10
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Storyshots Table Normal 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <table
    className="oec-table undefined"
    id="my-table"
  >
    <thead>
      <tr>
        <th
          className="oec-table__column-header oec-sortable"
          scope="col"
        >
          <span
            className="oec-table__column-title"
            onClick={[Function]}
          >
            Name
          </span>
          <div
            className="oec-table__sort-controls"
          >
            <input
              checked={true}
              id="my-table-col0-asc"
              name="my-table-column-0-sort"
              onChange={[Function]}
              type="checkbox"
              value="asc"
            />
            <label
              className="oec-table__sort-controls__asc"
              htmlFor="my-table-col0-asc"
            >
              <span>
                Sort ascending
              </span>
            </label>
            <input
              checked={false}
              id="my-table-col0-desc"
              name="my-table-column-0-sort"
              onChange={[Function]}
              type="checkbox"
              value="desc"
            />
            <label
              className="oec-table__sort-controls__desc"
              htmlFor="my-table-col0-desc"
            >
              <span>
                Sort descending
              </span>
            </label>
          </div>
        </th>
        <th
          className="oec-table__column-header oec-sortable"
          scope="col"
        >
          <span
            className="oec-table__column-title"
            onClick={[Function]}
          >
            Price
          </span>
          <div
            className="oec-table__sort-controls"
          >
            <input
              checked={false}
              id="my-table-col1-asc"
              name="my-table-column-1-sort"
              onChange={[Function]}
              type="checkbox"
              value="asc"
            />
            <label
              className="oec-table__sort-controls__asc"
              htmlFor="my-table-col1-asc"
            >
              <span>
                Sort ascending
              </span>
            </label>
            <input
              checked={false}
              id="my-table-col1-desc"
              name="my-table-column-1-sort"
              onChange={[Function]}
              type="checkbox"
              value="desc"
            />
            <label
              className="oec-table__sort-controls__desc"
              htmlFor="my-table-col1-desc"
            >
              <span>
                Sort descending
              </span>
            </label>
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          Apple
        </th>
        <th
          scope="row"
        >
          $0.75
        </th>
      </tr>
      <tr>
        <th
          scope="row"
        >
          Avocado
        </th>
        <th
          scope="row"
        >
          $1.80
        </th>
      </tr>
      <tr>
        <th
          scope="row"
        >
          Pear
        </th>
        <th
          scope="row"
        >
          $1.10
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Storyshots Tag Default 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <span
    className="usa-tag"
  >
    I AM A TAG
  </span>
</div>
`;

exports[`Storyshots TextInput Default 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <div
    className="usa-form-group"
  >
    <label
      className="usa-label"
      htmlFor="Default-text-input"
    >
      Default text input
    </label>
    <input
      className="usa-input"
      id="Default-text-input"
      name="Default-text-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
    />
  </div>
</div>
`;

exports[`Storyshots TextInput Disabled 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <div
    className="usa-form-group"
  >
    <label
      className="usa-label"
      htmlFor="Default-text-input"
    >
      Default text input
    </label>
    <input
      className="usa-input"
      disabled={true}
      id="Default-text-input"
      name="Default-text-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
    />
  </div>
</div>
`;

exports[`Storyshots TextInput Error 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <div
    className="usa-form-group usa-form-group--error"
  >
    <label
      className="usa-label usa-label--error"
      htmlFor="Default-text-input"
    >
      Default text input
    </label>
    <span
      className="usa-error-message"
      id="input-error-message"
      role="alert"
    >
      Fix this error
    </span>
    <input
      aria-describedby="input-error-message"
      className="usa-input usa-input--error"
      id="Default-text-input"
      name="Default-text-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
    />
  </div>
</div>
`;

exports[`Storyshots TextInput Success 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <div
    className="usa-form-group"
  >
    <label
      className="usa-label"
      htmlFor="Default-text-input"
    >
      Default text input
    </label>
    <input
      className="usa-input usa-input--success"
      id="Default-text-input"
      name="Default-text-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
    />
  </div>
</div>
`;

exports[`Storyshots TextInput With default value 1`] = `
<div
  style={
    Object {
      "padding": "40px",
    }
  }
>
  <div
    className="usa-form-group"
  >
    <label
      className="usa-label"
      htmlFor="Default-text-input"
    >
      Default text input
    </label>
    <input
      className="usa-input"
      defaultValue="default text"
      id="Default-text-input"
      name="Default-text-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
    />
  </div>
</div>
`;
